opentelemetry-sdk/tests/logs/test_export.py

```diff
@@ -342,6 +342,29 @@ def test_emit_call_log_record(self):
         logger.error("error")
         self.assertEqual(log_record_processor.emit.call_count, 1)
 
+    def test_logging_lib_not_invoked_in_batch_log_record_emit(self):  # pylint: disable=no-self-use
+        exporter = Mock()
+        processor = BatchLogRecordProcessor(exporter)
+        logger_provider = LoggerProvider(
+            resource=SDKResource.create(
+                {
+                    "service.name": "shoppingcart",
+                    "service.instance.id": "instance-12",
+                }
+            ),
+        )
+        logger_provider.add_log_record_processor(processor)
+        handler = LoggingHandler(
+            level=logging.INFO, logger_provider=logger_provider
+        )
+        # Attach OTLP handler to SDK logger
+        logging.getLogger("opentelemetry.sdk").addHandler(handler)
+        # If `emit` calls logging.log then this test will throw a maximum recursion depth exceeded exception and fail.
+        processor.emit(EMPTY_LOG)
+        processor.shutdown()
+        processor.emit(EMPTY_LOG)
+        logging.getLogger("opentelemetry.sdk").removeHandler(handler)
```

Comment:
sorry to be nitpicky but this won't run if there is an exception and it could cause unrelated tests to fail as red herrings since it swallows log messages.

can you add a contextmanager or try/finally block?